napari/utils/io.py

```diff
@@ -291,13 +292,54 @@ def write_csv(
             writer.writerow(row)
 
 
+def csv_to_layer_data(filename: str) -> List[FullLayerData]:
+    """Return layer data from a csv.
+
+    Parameters
+    ----------
+    filename : str
+        Filename for reading csv.
+
+    Returns
+    -------
+    list of napari.types.FullLayerData or None
+        Layer full layer data or None.
+    """
+
+    gen = iter_csv(filename)
+    column_names = next(gen)
+    layer_type = guess_layer_type_from_column_names(column_names)
+    if layer_type in csv_reader_functions:
+        return [csv_reader_functions[layer_type](filename)]
+    else:
+        return []
```

Comment:
Just for my own understanding, this function will conceivably be modified in the future to be able to read arbitrary csvs? It might be useful to e.g. be able to read a table of numbers (no headers whatsoever) as a coordinates table.

Reply:
yeah we could do that in a future PR